tasks/main.yml

```diff
@@ -1,6 +1,23 @@
 ---
 
-- name: Updating cache
+- name: Check server uptime
+  shell: cat /proc/uptime
+  register: host_uptime
+  changed_when: false
+  tags:
+    - packages
+
+- name: Update cache if server has been up less than an hour
+  become: yes
+  apt:
+    update_cache: yes
+    # Make idempotency test happy
+    cache_valid_time: 60
```

Comment:
The addition of `cache_valid_time` feels sorta brittle and odd. It's biased to pass the test, but would fail if the test waited a duration of time before running again. I don't have a solution for this however. Perhaps just use the same default variable as the other? Ie; `{{ cache_valid_time }}`

Reply:
I agree that this is hacky, but I can't bump it; we would run into the same issue. 
The only way I can see to get around this is to do a check to see if the file modified time of `/var/lib/apt/periodic/update-success-stamp` is very close to the server uptime. (I don't think we can count on it being the exact same time) If the file has been modified recently, we can assume that we don't need to force the update.